src/redux/books/books.js

```diff
@@ -0,0 +1,42 @@
+import { ADD_BOOK, DELETE_BOOK } from '../../constants';
+
+const newBook = (action) => {
+  const { categorie, title, author } = action;
+  return {
+    title,
+    categorie,
+    author,
+    id: Math.floor(Math.random() * 999999),
+  };
+};
+
+const removeBook = (state = [], id) => (state.filter((book) => book.id !== id));
+
+// Reducer
+export default function booksReducer(state = [], action) {
+  let books = null;
+  switch (action.type) {
+    case ADD_BOOK:
+      books = [...state, newBook(action)];
+      return books;
+    case DELETE_BOOK:
+      books = removeBook(state, action.id);
+      return books;
+    default:
+      return state;
+  }
+}
+
+// Action Creators
+
+export const addBook = (categorie, title, author) => ({
+  type: ADD_BOOK,
+  categorie,
+  title,
+  author,
+});
+
+export const deleteBook = (id) => ({
+  type: DELETE_BOOK,
+  id,
+});
```

Comment:
- [ ] At this moment your changes don't have any effect on your project as adding and removing books. Please implement it to proceed with the next step. 
  - I should be able to add and remove books from your app. 😉

Reply:
Yes, I know that but can you check the requirement again because this activity is just about creating reducers and actions but there is no anything in the requirements that says to connect that with the application but the next activity will be about connecting redux with react